server/client/client.go

```diff
@@ -0,0 +1,125 @@
+package client
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/google/go-github/v31/github"
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+
+	"github.com/mattermost/mattermost-plugin-github/server/plugin"
+)
+
+type PluginAPI interface {
+	PluginHTTP(*http.Request) *http.Response
+}
+
+type Client struct {
+	httpClient http.Client
+}
+
+type pluginAPIRoundTripper struct {
+	api PluginAPI
+}
+
+func (p *pluginAPIRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	resp := p.api.PluginHTTP(req)
+	if resp == nil {
+		return nil, errors.Errorf("Failed to make interplugin request")
+	}
+
+	return resp, nil
+}
+
+func NewClientPlugin(api PluginAPI) *Client {
```

Comment:
Does it make sense to call this `NewPluginClient`?

Reply:
:+1: from me. @crspeller what do you think?

Reply:
Ya, that sounds better.

Reply:
Changed :+1: